### ISSUE TITLE
Add example of how to read or write a slice of a dataset

### ISSUE THREAD

**Issue by jamesavery** [DavidAce/h5pp#3]
All the current examples show the use of readDataset / writeDataset to read in the full HDF5 file into memory in one go, or to write out the complete dataset from one allocated chunk of memory. 

However, a really common use of HDF5 is to work with large datasets, for which loading the whole dataset into memory is impossible. So the most common work-flow reads, processes, and writes only a slice or chunk of the dataset at a time. In h5py, one would write something like

```
voxels = h5in['voxels']   
result = h5out['result']  

for z0 in range(0,Nz,chunk_length):
   z1 = min(Nz, z0+chunk_length)
   result[z0:z1] = process(voxels[z0:z1])
```
which never loads more than chunk_length*np.prod(voxels.shape[1:]) elements into memory at any time.

It would be extremely useful if you would include an example of how to do the equivalent in h5pp - I tried looking at the header files, but could still only find out how to read or write a full dataset at once (which, when your dataset is hundreds of gigabytes, or even in the terabyte range, is just a no-go). 

I would love to use h5pp, as it seems like a fantastic project: the closest I've found to the ease of h5py for c++.

Thanks so much!

**Comment by DavidAce** [comment]
Thanks for the feedback! Reading and writing hyperslabs is already possible but I haven't added an example simply because of time constraints. But you are right, it's a central feature of HDF5 so I guess it's time I add such an example. Shouldn't take long at all.

**Comment by jamesavery** [comment]
That sounds excellent! A simple example of would be extremely useful, and make it possible for me to start moving some processing to c++ that is much easier to do efficiently there than in Python (by orders of magnitude).

**Comment by DavidAce** [comment]
The new examples (08a-08e) along with some minor changes are now up in the master branch. I think the interface for using hyperslabs is pretty simple but not to the level of h5py. I envy the indexing capabilities of python! But please tell me if the examples are wrong or unclear, or if you have any thoughts on how usage could be made simpler. All these changes will be included in the upcoming release of v1.8.5.

**Comment by DavidAce** [comment]
The examples are now included in the new release of h5pp v1.8.5. You are welcome to open a new issue (or discussion) if you have any other questions or feedback.
